src/Events/Controllers/EventsController.cs

```diff
@@ -63,6 +77,97 @@ public async Task<ActionResult<string>> Post([FromBody] CloudEvent cloudEvent)
             }
         }
 
+        /// <summary>
+        /// Retrieves a set of events related based on query parameters.
+        /// </summary>
+        /// <param name="after" example="3fa85f64-5717-4562-b3fc-2c963f66afa6">Retrieve events that were registered after this event Id</param>
+        /// <param name="source" example="[&quot;https://ttd.apps.at22.altinn.cloud/ttd/apps-test/&quot;
+        /// , &quot;https://ttd.apps.at22.altinn.cloud/digdir/bli-tjenesteeier/&quot;, &quot;https://ttd.apps.at22.altinn.cloud/ttd/apps-%/&quot;]">
+        /// A list of the sources to include</param>
+        /// <param name="type" example="[&quot;app.instance.created&quot;, &quot;app.instance.process.completed&quot;]">
+        /// A list of the event types to include</param>
+        /// <param name="subject">Optional filter by subject. Only exact matches will be returned.</param>
+        /// <param name="size">The maximum number of events to include in the response.</param>
+        [HttpGet()]
+        [Consumes("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Authorize(Policy = AuthorizationConstants.SCOPE_EVENTS_SUBSCRIBE)]
+        [Produces("application/cloudevents+json")]
+        public async Task<ActionResult<List<CloudEvent>>> Get(
+            [FromQuery] string after,
+            [FromQuery] List<string> source,
+            [FromQuery] List<string> type,
+            [FromHeader] string subject,
+            [FromQuery] int size = 50)
+        {
+            (bool isValid, string errorMessage) = ValidateQueryParams(after, size);
```

Comment:
How do we validate that a source value has been provided?

Reply:
Validation of source added in a more recent commit.